xdggs/index.py

```diff
@@ -34,20 +37,20 @@ def decode(ds, grid_info=None, *, name="cell_ids"):
     xarray.Dataset.dggs.decode
     xarray.DataArray.dggs.decode
     """
-    return ds.dggs.decode(name=name, grid_info=grid_info)
+    return ds.dggs.decode(name=name, grid_info=grid_info, index_kind=index_kind)
 
 
 class DGGSIndex(Index):
     _dim: str
-    _pd_index: PandasIndex
+    _index: PandasIndex
```

Comment:
It is a bit confusing (and potentially problematic) to define `_index` as a `PandasIndex` here, since in subclasses this might not be the case (i.e., HealpixIndex wrapping HealpixMocIndex).

Instead of `PandasIndex` this might just be `xr.Index`. Or maybe make it a generic type?

Reply:
I think I just missed that one, I've been changing the other declarations to `xr.Index`.

```suggestion
    _index: xr.Index
```